pkg/chains/signing/kms/kms.go

```diff
@@ -51,11 +52,35 @@ func NewSigner(ctx context.Context, cfg config.KMSSigner) (*Signer, error) {
 			return nil, err
 		}
 
-		conn, err := net.DialTimeout("tcp", vaultAddress.Host, 5*time.Second)
-		if err != nil {
-			return nil, err
+		var vaultUrl *url.URL
+		switch {
+		case vaultAddress.Port() != "":
+			vaultUrl = vaultAddress
+		case vaultAddress.Scheme == "http":
+			vaultUrl = &url.URL{
+				Scheme: vaultAddress.Scheme,
+				Host:   vaultAddress.Host + ":80",
+			}
+		case vaultAddress.Scheme == "https":
+			vaultUrl = &url.URL{
+				Scheme: vaultAddress.Scheme,
+				Host:   vaultAddress.Host + ":443",
+			}
+		case vaultAddress.Scheme == "":
+			vaultUrl = &url.URL{
+				Scheme: "http",
+				Host:   cfg.Auth.Address + ":80",
+			}
+		}
+
+		if vaultUrl != nil {
```

Comment:
Shouldn't we return an error if it is `nil`?

Reply:
as the error will be nil right ?

Reply:
Right. You want to create a new error in this case.